code/modules/materials/Mat_ProcsDefines.dm

```diff
@@ -190,19 +189,34 @@ var/global/list/triggerVars = list("triggersOnBullet", "triggersOnEat", "trigger
 	src.onMaterialChanged()
 
 /// sets the *appearance* of a material, but does not trigger any tiggerOnAdd or onMaterialChanged behaviour
+/// Order of precedence is as follows:
+/// if the material is in the list of appearences to ignore, do nothing
+/// If an iconstate exists in the icon for materialID-iconstate, that is chosen
+/// If the material has mat_changeappaerance set, then first texture is applied, then color (including alpha)
 /atom/proc/setMaterialAppearance(var/datum/material/mat1)
-	var/set_color_alpha = TRUE
 	src.alpha = 255
 	src.color = null
-	src.UpdateOverlays(null, "material")
-	if (islist(src.mat_appearances_to_ignore) && length(src.mat_appearances_to_ignore))
-		if (mat1.name in src.mat_appearances_to_ignore)
-			set_color_alpha = FALSE
-	if (set_color_alpha && src.mat_changeappearance && mat1.applyColor)
+	if (length(src.mat_appearances_to_ignore) && (mat1.name in src.mat_appearances_to_ignore))
+		return
+
+	var/potential_new_icon_state = "[mat1.mat_id]-[initial(src.icon_state)]"
```

Comment:
While this made sense for the bars I wonder if in the general case the opposite order of these two things doesn't make more sense?

Reply:
Also probably should use something like "icon_state but with the possible material thingie stripped" instead of initial().

I wonder if it might make sense to choose some unique separator of the icon state and the mat id here. Like bar$gold or something

Reply:
oh you thinking like splitting the current icon_state on $ and replacing the prefix with materialID? I can do that

Reply:
Yeah but as I mentioned maybe the mat id should be a suffix idk. Seems more readable to me to put the "base" state first and the material second.